### DISCUSSION TITLE
How to compile the assets to the VULKAN API using Harfang's asset compiler?

### DISCUSSION THREAD

**Discussion by Pyblek** [harfang3d/harfang3d#47] [Q&A] [answered]
Hi,

I'mtrying to compile my graphics assets using assetc.exe (running Win10 64bits). The command line seems to be valid but I only get compilation errors.

Here is my commmand line :

```bash
assetc.exe resources -api VULKAN
```

first I get this feedback, letting me think everything will be ok :

```
Harfang ASSETC 1.1
(4589:363:518:900) > Input dir: resources
(4589:363:659:500) > Output dir: resources_compiled
(4589:364:198:900)
(4589:364:655:800) > Target platform: windows
(4589:365:188:500) > Target graphics API: VULKAN
(4589:365:712:100) > Target pipeline: forward
```

but then it reports a lot of errors, as if none of the assets could be compiled :

```
    557 input files
    1240 output files
    0 processed
    585 failed

  Saving compilation DB 'resources_compiled/assetc.cab'
Compilation done, took 965ms
```

what am I missing, here?

**Comment by astrofra** [comment, marked as answer]
Hi!

The correct syntax for the API switch is either `DX11`, `DX12`, `GL`, `GLES` or `VK`. What you are looking for is `VK` that stands for **Vulkan**.

So, the command line should read like that :

```bash
assetc resources -api VK
```
